### DISCUSSION TITLE
Multi color prints: Which approach are you using?

### DISCUSSION THREAD

**Discussion by alphatrianguli** [Jyers/Marlin#612] [Q&A] [answered]
This is not plainly a FW question, but could be:

What approach are you guys using, when doing multi-color prints? I know there is the CURA option to pause at a certain height - but is there another option, related to the FW itself?

Thanks for your inputs!

**Comment by t3chguy** [comment, marked as answer]
The firmware supports M600 which will automatically pause, park the tool head, eject filament, wait for you to remove filament, insert new filament, click OK on the screen and it'll re-heat, re-load filament and then continue the print.

This M600 is also possible from the UI during a printer as `Change Filament`

**Reply by alphatrianguli** [reply]
> The firmware supports M600 which will automatically pause, park the tool head, eject filament, wait for you to remove filament, insert new filament, click OK on the screen and it'll re-heat, re-load filament and then continue the print.
> 
> This M600 is also possible from the UI during a printer as `Change Filament`

Wow, this is nice, didnt know that! :) So basically, I can put a M600 in my GCODE at the place I want to pause for "Change Filament"? Would I have to add this manually? Or is there a plug-in in CURA? Or how would you / one do this most easily?

Thanks!

**Reply by t3chguy** [reply]
No idea about Cura but PrusaSlicer has built in support for it and does it when you make use of PS's multi colour features

**Reply by Jyers** [reply]
Cura has "post processing scripts" which you can use to add M600 commands at a certain height or layer

**Reply by alphatrianguli** [reply]
> Cura has "post processing scripts" which you can use to add M600 commands at a certain height or layer

Thanks for the input. I've tried the "Pause at Height" plug-in before but it never worked properly. Furthermore I think it doesnt do a M600. So I would use the "Change Filament" post processing plug-in?

**Reply by Jyers** [reply]
Yes, either that or I believe there are some third party scripts that allow a little more customization

**Reply by alphatrianguli** [reply]
Tried today and its working flawlessly - thanks a lot! :)

The CURA plug-in though adds the following line:

M600 E30.00 L300.00 X0.00 Y0.00 ;

Are those settings even nessecary or are they overwritten anyway by the FW itself? :)

And something else: After the nozzle is heated up again - the filament is extruded and retracted several times - is that normal? It almost looks like the filament is skipping through the gears...

**Reply by Jyers** [reply]
M600 does some retractions and such to try to improve the unload and load. Also I don't think the firmware will override those settings, but if you do have them set in the firmware you don't really need those extra arguments

**Reply by alphatrianguli** [reply]
I see. Then I will change the plugin-script to just do an M600. Thanks very much for your support Jake!
